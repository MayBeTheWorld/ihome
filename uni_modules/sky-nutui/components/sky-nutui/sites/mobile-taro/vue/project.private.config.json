{
  "setting": {
    "compileHotReLoad": true
  },
  "description": "项目私有配置文件。此文件中的内容将覆盖 project.config.json 中的相同字段。项目的改动优先同步到此文件中。详见文档：https://developers.weixin.qq.com/miniprogram/dev/devtools/projectconfig.html",
  "condition": {
    "miniprogram": {
      "list": [
        {
          "name": "feedback/pages/notify/index",
          "pathName": "feedback/pages/notify/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "pages/index/index",
          "pathName": "pages/index/index",
          "query": "",
          "scene": null
        },
        {
          "name": "exhibition/pages/countdown/index",
          "pathName": "exhibition/pages/countdown/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/collapse/index",
          "pathName": "feedback/pages/collapse/index",
          "query": "",
          "scene": null
        },
        {
          "name": "business/pages/signature/index",
          "pathName": "business/pages/signature/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "exhibition/pages/collapse/index",
          "pathName": "exhibition/pages/collapse/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "dentry/pages/textarea/index",
          "pathName": "dentry/pages/textarea/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        }
      ]
    }
  },
  "projectname": "NutUI-Taro"
}