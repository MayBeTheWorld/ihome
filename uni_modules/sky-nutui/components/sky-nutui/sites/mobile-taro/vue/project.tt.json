{
  "miniprogramRoot": "./",
  "projectname": "@nutui/nutui-taro-mobile",
  "appid": "touristappid",
  "setting": {
    "es6": false,
    "minified": false
  }
}
