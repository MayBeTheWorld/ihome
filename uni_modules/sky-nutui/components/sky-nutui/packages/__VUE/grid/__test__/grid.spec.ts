import { h, nextTick } from 'vue';
import { config, mount } from '@vue/test-utils';
import { useRouter } from 'vue-router';
import Grid from '../index.vue';
import GridItem from '../../griditem/index.vue';
import NutIcon from '../../icon/index.vue';

beforeAll(() => {
  config.global.components = {
    NutIcon
  };
});

afterAll(() => {
  config.global.components = {};
});

// mock module
jest.mock('vue-router', () => ({
  useRouter: jest.fn()
}));

test('should render square correctly', () => {
  const wrapper = mount(Grid, {
    props: {
      square: true,
      columnNum: 2
    },
    slots: {
      default: [GridItem, GridItem, GridItem, GridItem]
    }
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should render gutter correctly', () => {
  const wrapper = mount(Grid, {
    props: {
      gutter: 20
    },
    slots: {
      default: [GridItem, GridItem, GridItem, GridItem, GridItem, GridItem, GridItem, GridItem]
    }
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should change icon and color when using icon-size and icon-color prop', () => {
  const wrapper = mount(Grid, {
    props: {
      iconSize: 30
    },
    slots: {
      default: h(GridItem, {
        iconColor: 'red'
      })
    }
  });

  expect(wrapper.find<HTMLElement>('.nut-icon').element.style.fontSize).toEqual('30px');
  expect(wrapper.find<HTMLElement>('.nut-icon').element.style.color).toEqual('red');
});

test('should render default slot correctly', () => {
  const wrapper = mount(Grid, {
    slots: {
      default: h(GridItem, null, {
        default: () => 'Default Slot'
      })
    }
  });

  expect(wrapper.find('.nut-grid-item__content').html()).toContain('Default Slot');
  expect(wrapper.html()).toMatchSnapshot();
});

test('should emit click correctly', async () => {
  const wrapper = mount(Grid, {
    slots: {
      default: [GridItem]
    }
  });

  wrapper.find('.nut-grid-item').trigger('click');
  await nextTick();

  expect(wrapper.emitted('click')).toHaveLength(1);
});

test('should navifation correctly', async () => {
  // 当 `useRouter()` 时返回 `push` 方法
  const push = jest.fn((url: string) => url);
  (useRouter as jest.Mock).mockImplementationOnce(() => ({
    push
  }));

  const wrapper = mount(Grid, {
    slots: {
      default: h(GridItem, {
        to: 'http://m.jd.com'
      })
    }
  });
  expect(wrapper.html()).toMatchSnapshot();
  // wrapper.find('.nut-grid-item').trigger('click');
  // await nextTick();
  // expect(push.mock.calls[0][0]).toEqual('/home');
});
