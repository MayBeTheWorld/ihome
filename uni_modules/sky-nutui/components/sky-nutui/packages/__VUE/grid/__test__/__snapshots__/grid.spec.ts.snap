// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should navifation correctly 1`] = `
"<view class=\\"nut-grid nut-grid--border\\">
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 25%;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--center nut-grid-item__content--clickable\\"><i class=\\"nutui-iconfont nut-icon nut-icon-\\" style=\\"font-size: 28px; width: 28px; height: 28px;\\" src=\\"\\"></i>
      <view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
</view>"
`;

exports[`should render default slot correctly 1`] = `
"<view class=\\"nut-grid nut-grid--border\\">
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 25%;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--center\\">Default Slot</view>
  </view>
</view>"
`;

exports[`should render gutter correctly 1`] = `
"<view class=\\"nut-grid\\" style=\\"padding-left: 20px;\\">
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 25%; padding-right: 20px;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--surround nut-grid-item__content--center\\"><i class=\\"nutui-iconfont nut-icon nut-icon-\\" style=\\"font-size: 28px; width: 28px; height: 28px;\\" src=\\"\\"></i>
      <view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 25%; padding-right: 20px;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--surround nut-grid-item__content--center\\"><i class=\\"nutui-iconfont nut-icon nut-icon-\\" style=\\"font-size: 28px; width: 28px; height: 28px;\\" src=\\"\\"></i>
      <view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 25%; padding-right: 20px;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--surround nut-grid-item__content--center\\"><i class=\\"nutui-iconfont nut-icon nut-icon-\\" style=\\"font-size: 28px; width: 28px; height: 28px;\\" src=\\"\\"></i>
      <view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 25%; padding-right: 20px;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--surround nut-grid-item__content--center\\"><i class=\\"nutui-iconfont nut-icon nut-icon-\\" style=\\"font-size: 28px; width: 28px; height: 28px;\\" src=\\"\\"></i>
      <view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 25%; padding-right: 20px; margin-top: 20px;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--surround nut-grid-item__content--center\\"><i class=\\"nutui-iconfont nut-icon nut-icon-\\" style=\\"font-size: 28px; width: 28px; height: 28px;\\" src=\\"\\"></i>
      <view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 25%; padding-right: 20px; margin-top: 20px;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--surround nut-grid-item__content--center\\"><i class=\\"nutui-iconfont nut-icon nut-icon-\\" style=\\"font-size: 28px; width: 28px; height: 28px;\\" src=\\"\\"></i>
      <view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 25%; padding-right: 20px; margin-top: 20px;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--surround nut-grid-item__content--center\\"><i class=\\"nutui-iconfont nut-icon nut-icon-\\" style=\\"font-size: 28px; width: 28px; height: 28px;\\" src=\\"\\"></i>
      <view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 25%; padding-right: 20px; margin-top: 20px;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--surround nut-grid-item__content--center\\"><i class=\\"nutui-iconfont nut-icon nut-icon-\\" style=\\"font-size: 28px; width: 28px; height: 28px;\\" src=\\"\\"></i>
      <view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
</view>"
`;

exports[`should render square correctly 1`] = `
"<view class=\\"nut-grid nut-grid--border\\">
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 50%; padding-top: 50%;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--center nut-grid-item__content--square\\"><i class=\\"nutui-iconfont nut-icon nut-icon-\\" style=\\"font-size: 28px; width: 28px; height: 28px;\\" src=\\"\\"></i>
      <view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 50%; padding-top: 50%;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--center nut-grid-item__content--square\\"><i class=\\"nutui-iconfont nut-icon nut-icon-\\" style=\\"font-size: 28px; width: 28px; height: 28px;\\" src=\\"\\"></i>
      <view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 50%; padding-top: 50%;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--center nut-grid-item__content--square\\"><i class=\\"nutui-iconfont nut-icon nut-icon-\\" style=\\"font-size: 28px; width: 28px; height: 28px;\\" src=\\"\\"></i>
      <view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 50%; padding-top: 50%;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--center nut-grid-item__content--square\\"><i class=\\"nutui-iconfont nut-icon nut-icon-\\" style=\\"font-size: 28px; width: 28px; height: 28px;\\" src=\\"\\"></i>
      <view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
</view>"
`;
